### PR TITLE
Migrate API integration tests and refactor them

### LINKED ISSUE #17441: Migrate API integration tests and refactor them
## Description
API integration tests will be moved to the wazuh/wazuh repository. However, they must be refactored to improve multiple aspects.

### Current behavior
Some tests are failing. In addition, the old tag's approach is used so a lot of tests are skipped. More things could arise after being analyzed in this issue:
- https://github.com/wazuh/wazuh/issues/17157

### Expected behavior
- There should be no skipped tests.
- Any failing test reported at https://github.com/wazuh/wazuh/issues/17157 should be fixed.
- Global fixtures and tools should be used when possible (for example, to restart services).
- Review whether the structure used in tests like `test_vulnerability_detector` could be applied for the API (configuration template, test_cases, etc).

### QA framework migration

This issue requires some code from the old QA framework to be moved to the new repository. So, the following issue: https://github.com/wazuh/qa-integration-framework/issues/6 contains those changes that support the migration of THESE tests.

### PR DESCRIPTION
**Pull request by mauromalara**
|Related issue|
|---|
| closes #17441 |

## Description

Move API integration tests to the `wazuh/wazuh` repository, refactor code, and clean unused/duplicated code.

## Execution update

| R1 | R2 | R3 | OS | Commit | Notes
|--|--|--|--|--|--
| [🟢](https://github.com/wazuh/wazuh/files/11911140/Archive.zip) | [🟢](https://github.com/wazuh/wazuh/files/11911141/Archive.2.zip) | [🟢](https://github.com/wazuh/wazuh/files/11911139/Archive.3.zip) | Ubuntu 22 (Docker) | ce0336f | -

### CODE REVIEW

**Comment by mauromalara** [comment]
### PR to remove the execution of the tests from PR_wazuh

https://github.com/wazuh/wazuh-jenkins/pull/5345

File: .github/workflows/integration-tests-api-tier-0-1.yml [hunk 1]
```diff
@@ -0,0 +1,71 @@
+name: Integration tests for API - Tier 0 and 1
+
+on:
+  pull_request:
+    paths:
+        - ".github/workflows/integration-tests-api-tier-0-1.yml"
```
Comment (Selutario) [inline .github/workflows/integration-tests-api-tier-0-1.yml:6]: When new changes are added inside `framework/wazuh/rbac/`, `test_rbac` should also be run.

Comment (mauromalara) [reply .github/workflows/integration-tests-api-tier-0-1.yml:6]: Done in new PR (due to duplicated commits in this branch)

**Comment by mauromalara** [comment]
This development was moved to this PR because some of the commits were duplicated due to a bad rebase. https://github.com/wazuh/wazuh/pull/17733

**Comment by mauromalara** [comment]
This development was moved to this PR because some of the commits were duplicated due to a bad rebase. https://github.com/wazuh/wazuh/pull/17733

**Pull request closed without merging**